# HG changeset patch
# User Jaroslav Hajek <highegg@gmail.com>
# Date 1246706360 -7200
#      Sat Jul 04 13:19:20 2009 +0200
# Node ID 0dd3c7a2ba19af3eaec6053b49dbe0d048fc31aa
# Parent  f008a3a1bcb05635fb7c3abfc5918cb8eaf92456
undo escaped change in previous patch

diff --git a/src/ov-fcn-handle.cc b/src/ov-fcn-handle.cc
--- a/src/ov-fcn-handle.cc
+++ b/src/ov-fcn-handle.cc
@@ -1344,17 +1344,20 @@ make_fcn_handle (const std::string& nm)
 	case '|':
 	  tnm = "or";
 	  break;
 	}
     }
 
   octave_value f = symbol_table::find_function (tnm);
 
-  retval = octave_value (new octave_fcn_handle (f, tnm));
+  if (f.is_defined ())
+    retval = octave_value (new octave_fcn_handle (f, tnm));
+  else
+    error ("error creating function handle \"@%s\"", nm.c_str ());
 
   return retval;
 }
 
 /*
 %!test
 %! x = {".**", "power";
 %!      ".'", "transpose";
